src/wireviz/wv_html.py

```diff
@@ -2,54 +2,86 @@
 # -*- coding: utf-8 -*-
 
 from pathlib import Path
-from typing import List, Union
+from typing import Dict, List, Union
 import re
 
 from wireviz import __version__, APP_NAME, APP_URL, wv_colors
 from wireviz.DataClasses import Metadata, Options
-from wireviz.wv_helper import flatten2d, open_file_read, open_file_write
+from wireviz.wv_helper import flatten2d, open_file_read, open_file_write, smart_file_resolve
+from wireviz.wv_gv_html import html_line_breaks
 
 def generate_html_output(filename: Union[str, Path], bom_list: List[List[str]], metadata: Metadata, options: Options):
+
+    # load HTML template
+    if 'name' in metadata.get('template',{}):
+        # if relative path to template was provided, check directory of YAML file first, fall back to built-in template directory
+        templatefile = smart_file_resolve(f'{metadata["template"]["name"]}.html', [Path(filename).parent, Path(__file__).parent / 'templates'])
+    else:
+        # fall back to built-in simple template if no template was provided
+        templatefile = Path(__file__).parent / 'templates/simple.html'
```

Comment:
Or a compact alternative:
```suggestion
    templatename = metadata.get('template',{}).get('name')
    templatefile = smart_file_resolve(f'{templatename or "simple"}.html',
        [Path(filename).parent, Path(__file__).parent / 'templates'][not templatename:])
```

Reply:
The compact alternative looks nice but leaves one (admittedly improbable) edge case:
- No template name is specified
  - Expectation: User expects to fall back to the WireViz built-in template
- There is a `simple.html` file in one of the paths to resolve
  - Actual behavior: this `simple.html` is used instead of the built-in one.

Reply:
> The compact alternative looks nice but leaves one (admittedly improbable) edge case, if no template name is specified in the YAML, but there is a `simple.html` file in one of the paths to resolve.
> 
> **Expectation**: User expects to fall back to the WireViz built-in template
> **Actual behavior**: The `simple.html` found by `smart_file_resolve()` is used instead of the built-in one.
> 
> Also, I suppose the `[not templatename:]` at the end of line 18 is a typo?

Actually, the list slicing at the end of line 18 is not a typo, but removes the first element of the list when no template name is specified (`[True:]` equals `[1:]`), and that takes care of your edge case above. When a template name _is_ specified (`[False:]` equals `[0:]`), both elements of the list is used.

Maybe an additional comment below line 18 might help:
```py
        # The first path in the list above is sliced off when no template name is specified.
```

Reply:
In case the metadata is specified in a prepended YAML file (probably unlikely until we support overriding individual metadata elements from the main YAML file), maybe the paths of both files should be collected in a list and passed into a new parameter of `generate_html_output()`. In that case, I suggest this variation:
```suggestion
    templatename = metadata.get('template',{}).get('name')
    templatefile = smart_file_resolve(f'{templatename or "simple"}.html',
        [*input_paths, Path(__file__).parent / 'templates'][-(not templatename):])
        # All but the last path in the list above is sliced off when no template name is specified.
```

Reply:
Sorry, these alternatives do not feel very pythonic and I would rather keep it as it is.